test/meta/generate_table_init.js

```diff
@@ -80,26 +81,31 @@ emit_a();
 // to count through the vector entries when debugging.
 let e = undefined;
 
-// Passive init that overwrites all-null entries
-tab_test("(table.init 1 (i32.const 7) (i32.const 0) (i32.const 4))",
-         [e,e,3,1,4, 1,e,2,7,1, 8,e,7,5,2, 3,6,e,e,e, e,e,e,e,e, e,e,e,e,e]);
-
-// Passive init that overwrites existing active-init-created entries
-tab_test("(table.init 3 (i32.const 15) (i32.const 1) (i32.const 3))",
-         [e,e,3,1,4, 1,e,e,e,e, e,e,7,5,2, 9,2,7,e,e, e,e,e,e,e, e,e,e,e,e]);
-
-// Perform active and passive initialisation and then multiple copies
-tab_test(`(table.init 1 (i32.const 7) (i32.const 0) (i32.const 4))
-    (elem.drop 1)
-    (table.init 3 (i32.const 15) (i32.const 1) (i32.const 3))
-    (elem.drop 3)
-    (table.copy (i32.const 20) (i32.const 15) (i32.const 5))
-    (table.copy (i32.const 21) (i32.const 29) (i32.const 1))
-    (table.copy (i32.const 24) (i32.const 10) (i32.const 1))
-    (table.copy (i32.const 13) (i32.const 11) (i32.const 4))
-    (table.copy (i32.const 19) (i32.const 20) (i32.const 5))`,
-         [e,e,3,1,4, 1,e,2,7,1, 8,e,7,e,7, 5,2,7,e,9, e,7,e,8,8, e,e,e,e,e]);
-
+for ( let table of [0, 1] ) {
+    // Passive init that overwrites all-null entries
+    tab_test(`(table.init $t${table} 1 (i32.const 7) (i32.const 0) (i32.const 4))`,
```

Comment:
We recently clarified in the binary format that elem index comes before table index. Seems like we'll want to do the same for the text format too? If so, we may want to do that before landing this change.

Reply:
I would prefer to keep this as an oddity of the binary format. For the human-readable text form (and the AST) it's nicer if init matches the dst-src parameter order of copy and the operands of other instructions.

Reply:
(Why was this switched in the first place?)

Reply:
My guess is that, since the table/memory index was always 0, it made sense to put it at the end (e.g. similar to `call_indirect`)

Reply:
FWIW, I agree with @rossberg re the operand order in the text format.

One thing that annoys me about the text format though is that in the elem segment I must write `(table $t)` while in call_indirect, table.copy, table.init, etc, I can only write $t.  If the text format would admit disambiguating syntax it could also admit a more flexible operand order: `(table.init (elem 1) (table $t) ...)`.  Something for another day perhaps.

Reply:
Hm, that would be piggybacking on the wrong thing, since that syntax would only happen to work where the immediates are of different sort, e.g. not table.copy. Moreover, for various instructions, src and dst are denoted by a combination of immediates and stack operands, and their order is supposed to match.

Reply:
I agree that's true for the order of the table operands.  It's still annoying that the syntax is non-uniform, but it's just a hobbyhorse - not a problem to solve here.